community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/IndexReference.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.newapi;
+
+import org.neo4j.internal.kernel.api.IndexCapability;
+import org.neo4j.internal.kernel.api.IndexOrder;
+import org.neo4j.internal.kernel.api.IndexValueCapability;
+import org.neo4j.values.storable.ValueGroup;
+
+class IndexReference implements org.neo4j.internal.kernel.api.CapableIndexReference
+{
+    private final int label;
+    private final int[] properties;
+    private final boolean unique;
```

Comment:
It somehow feels wrong to create a new index meta class that is a `IndexDescriptor` + capabilities. Could we do composition instead? We don't have to fix it now, just venting opinion.

Reply:
Yes, I agree it's a bit awkward. From what we discussed yesterday we might want to split up the code path in two.
```
public interface SchemaRead
{
    IndexDescription index( int label, int... properties );

    IndexCapability indexCapability( IndexDescription indexDescription );
}
```
Let's address this issue at that point.